service/worker/replicator/replicationTask.go

```diff
@@ -0,0 +1,291 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package replicator
+
+import (
+	"context"
+	"time"
+
+	"github.com/dgryski/go-farm"
+
+	"github.com/uber/cadence/common"
+
+	"github.com/uber/cadence/.gen/go/replicator"
+
+	"github.com/uber-common/bark"
+	"github.com/uber/cadence/common/logging"
+	"github.com/uber/cadence/common/metrics"
+	"github.com/uber/cadence/common/xdc"
+
+	"github.com/uber/cadence/common/locks"
+
+	"github.com/uber/cadence/client/history"
+
+	h "github.com/uber/cadence/.gen/go/history"
+	"github.com/uber/cadence/.gen/go/shared"
+	"github.com/uber/cadence/common/definition"
+	"github.com/uber/cadence/common/messaging"
+)
+
+type (
+	workflowReplicationTask struct {
+		queueID  definition.WorkflowIdentifier
+		taskID   int64
+		attempt  int
+		kafkaMsg messaging.Message
+		logger   bark.Logger
+
+		config              *Config
+		historyClient       history.Client
+		metricsClient       metrics.Client
+		historyRereplicator xdc.HistoryRereplicator
+		resendLock          locks.IDMutex
+	}
+
+	activityReplicationTask struct {
+		workflowReplicationTask
+		req *h.SyncActivityRequest
+	}
+
+	historyReplicationTask struct {
+		workflowReplicationTask
+		req *h.ReplicateEventsRequest
+	}
+)
+
+const (
+	replicationTaskRetryDelay = 500 * time.Microsecond
+)
+
+func newActivityReplicationTask(task *replicator.ReplicationTask, msg messaging.Message, logger bark.Logger,
```

Comment:
nit: stacking the arguments makes it more readable.. 

```
func newActivityReplicationTask(
    task *replicator.ReplicationTask, 
    msg messaging.Message, 
    logger bark.Logger,
    config *Config, 
    historyClient history.Client, 
    metricsClient metrics.Client,
    historyRereplicator xdc.HistoryRereplicator,
) *activityReplicationTask {
```

Also, ordering of the params is weird. conf, metrics, logger should be next to each other. And conf is normally first or second for constructors.